src/qiboml/models/optimizers.py

```diff
@@ -0,0 +1,391 @@
+from qibo.models.encodings import hamming_weight_encoder
+from qibo import get_backend
+
+
+class ExactGeodesicTransportCG:
+    """Exact Geodesic Transport with Conjugate Gradients Optimizer.
+
+    Implements the Exact Geodesic Transport with Conjugate Gradients (EGT-CG) optimizer, a curvature-aware Riemannian optimizer designed
+    specifically for variational circuits based on the Hamming-weight encoder (HWE) ansätze. It updates parameters along exact geodesic paths
+    on the hyperspherical manifold defined by the HWE, combining analytic metric computation, conjugate-gradient memory, and dynamic
+    learning rates for fast, globally convergent optimization.
+
+    Args:
+      n_qubits (int): Number of qubits in the quantum circuit.
+      weight (int): Hamming weight to encode.
+      hamiltonian (qibo.hamiltonians.Hamiltonian): Hamiltonian whose expectation
+        value defines the loss to minimize.
+      theta (array): Initial spherical angles parameterizing the amplitudes.
+      backtrack_rate (float, optional): Backtracking rate for Wolfe condition
+        line search. Defaults to 0.5 if not provided.
+      geometric_gradient (bool, optional): If True, uses the geometric gradient
+        estimator instead of numerical finite differences. Defaults to False.
+      backend (qibo.backends.AbstractBackend, optional): Qibo backend to use for numerical
+        operations. If None, the default backend is used.
+      multiplicative_factor (float, optional): Scaling factor applied to the intial learning
+        rate during optimization. Defaults to 1.
+      c1 (float, optional): Constant for Armijo condition (sufficient decrease) in
+        Wolfe line search. Defaults to 1e-3.
+      c2 (float, optional): Constant for curvature condition in Wolfe line search.
+        hould satisfy c1 < c2 < 1. Defaults to 0.9.
+
+    Returns:
+      ExactGeodesicTransportCG: Instantiated optimizer object.
+
+    Adapted from: Ferreira‑Martins, A. J., Farias, R. M. S., Camilo, G., Maciel, T. O., Tosta, A., Lin, R., Alhajri, A., Haug, T., & Aolita, L. (2025, June 20).
+    Variational quantum algorithms with exact geodesic transport. arXiv:2506.17395
```

Comment:
For instance, see reference style in https://github.com/qiboteam/qibo/blob/3e304266cd7a9909482e03cbc131d9b27a914de1/src/qibo/models/encodings.py#L170